tests/integration/security/sds_ingress/single_mtls_gateway_separate_secret/single_mtls_gateway_separate_secret_test.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package singlemtlsgatewayseparatesecret
+
+import (
+	"time"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/ingress"
+	ingressutil "istio.io/istio/tests/integration/security/sds_ingress/util"
+
+	"testing"
+)
+
+var (
+	credName   = []string{"bookinfo-credential-1"}
+	credCaName = []string{"bookinfo-credential-1-cacert"}
+	host       = "bookinfo1.example.com"
+)
+
+// TestSingleMTLSGateway_ServerKeyCertRotation tests a single mTLS ingress gateway with SDS enabled.
+// Verifies behavior in these scenarios.
+// (1) create two kubernetes secrets to provision server key/cert and client CA cert, and
+// verify that mTLS connection could establish to deliver HTTPS request.
+// (2) replace kubernetes secret to rotate server key/cert, and verify that mTLS connection could
+// not establish because the new server key/cert does not match client CA cert.
```

Comment:
This description doesn't make sense to me. The client CA cert is used to by the server to verify the client. If the client cert used in the request doesn't change from before the server key/cert rotation, the request should still succeed (from the server validation). If this implies that the client should now reject the server then the wording could be better in that regard.

You check for a 200 response code. Are there error response codes to distinguish client cert failed validation (server doing the rejection) and server cert failed validation (client doing the rejection)?

Reply:
Thanks for pointing this out. I have updated the test comment.  The (2) case is client using old server CA cert to validate the mis-matched server cert, and the validation fails.
For a server cert validation failure, the error messages is "x509: certificate signed by unknown authority".
For a client cert validation failure, the error message is "remote error: tls: unknown certificate authority" 
The response error codes are 0 for both cases.